ha/setup/pre_disruptive_upgrade.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""Module to do pre-upgrade routines in scope of Disruptive Upgrade feature."""
+
+import argparse
+import os
+import shlex
+import subprocess
+import tarfile
+from glob import glob
+from shutil import copytree, rmtree
+from time import gmtime, strftime
+from typing import Any
+from xml.etree import ElementTree
+from xml.etree.ElementTree import Element
+
+from cortx.utils.log import Log
+from ha.const import (BACKUP_DEST_DIR_CONF, BACKUP_DEST_DIR_CONSUL,
+                      CONFIG_DIR, RA_LOG_DIR)
+from ha.execute import SimpleCommand
+
+
+class PreRequisiteUpgradeError(Exception):
```

Comment:
can we move this to error.py

Reply:
Moved to ha.core.error as HAError successor.